apps/api/src/app/events/usecases/trigger-event/message-filter.matcher.ts

```diff
@@ -1,24 +1,23 @@
 import { NotificationStepEntity } from '@novu/dal';
+import * as _ from 'lodash';
 
-export function matchMessageWithFilters(
-  steps: NotificationStepEntity[],
-  payloadVariables: { [key: string]: string | string[] | { [key: string]: string } }
-): NotificationStepEntity[] {
-  return steps.filter((message) => {
-    if (message.filters?.length) {
-      const foundFilter = message.filters.find((filter) => {
-        if (filter.type === 'GROUP') {
-          return handleGroupFilters(filter, payloadVariables);
-        }
-
-        return false;
-      });
+export function matchMessageWithFilters(step: NotificationStepEntity, payloadVariables: any): boolean {
+  if (!Array.isArray(step?.filters)) {
+    return true;
+  }
+  const firstFilter = step.filters[0]?.children;
+  if (!firstFilter || (Array.isArray(firstFilter) && firstFilter.length === 0)) {
+    return true;
+  }
```

Comment:
Could you eloborate here? Not sure I understand this part. Or can we make it so it works without accessing the '0' position ?

Reply:
That filters is an array is because we like to select multiple filter groups in the future like 'and' and 'or' 😄

Reply:
Oh wait I will refactor this...